### ISSUE TITLE
[foxfooding] Wrong modal displayed for sign in using existing firefox account

### ISSUE THREAD

**Issue by pdehaan** [mozilla/fxa#14059]
**Firefox Version:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:106.0) Gecko/20100101 Firefox/106.0
**Window Size (inner width and height):** 1236x970
**GitHub Username:** @siddhartharao17
## Steps to Reproduce
I already have a Firefox account that I have been using since 07/2018.

For the testing scenario "Existing Firefox Account",
1. I clicked "Sign In" in the application menu in the top-right corner of the browser
2. I put in my email address on the "Continue to Firefox accounts" page and clicked "Sign up or Sign in"
3. The page said "Set your password" and asked me to set a password and confirm age. 

## Expected Behavior
After Step #2 above, I was expecting the page to ask me for the password for my existing account, i.e. I was expecting the "Enter your password" modal to be displayed.

## Actual Behavior
For the testing scenario "Existing Firefox Account",
1. I clicked "Sign In" in the application menu in the top-right corner of the browser
2. I put in my email address on the "Continue to Firefox accounts" page and clicked "Sign up or Sign in"
3. The page said "Set your password" and asked me to set a password and confirm age. 

In summary, sign up modal was displayed when I was expecting a sign in modal.

## Attachment
![Attachment](https://mk-mozilla-campaign-form.s3.eu-central-1.amazonaws.com/c2444c08-54af-48df-a907-c8f02486ce68.png)
[Link to the original attachment](https://mk-mozilla-campaign-form.s3.eu-central-1.amazonaws.com/c2444c08-54af-48df-a907-c8f02486ce68.png)

**Comment by siddhartharao17** [comment]
actual vs expected
![Screen Shot 2022-09-02 at 15 52 49](https://user-images.githubusercontent.com/8169998/188229700-8dd44731-8f6f-4aff-b957-578952cd6df9.png)
![Screen Shot 2022-09-02 at 15 53 21](https://user-images.githubusercontent.com/8169998/188229706-05564d20-a0ad-4203-b718-ba87089be9ea.png)

**Comment by siddhartharao17** [comment]
actual vs expected

![Screen Shot 2022-09-02 at 15 52 49](https://user-images.githubusercontent.com/8169998/188229909-7ea4ad11-ab88-4ae8-ad09-378f6e3d9cf3.png)
![Screen Shot 2022-09-02 at 15 53 21](https://user-images.githubusercontent.com/8169998/188229912-b14ec3e6-cb2e-4d33-8b48-51f187d9dab4.png)

**Comment by clouserw** [comment]
@siddhartharao17 did you have a primary account at one point and swap the primary and secondary email addresses or anything similar to that?

**Comment by siddhartharao17** [comment]
@clouserw No, I don't think so. This has been my only Firefox account. The sign in works as expected on FF 104.0.1 though. The "Expected" screenshot is from the login I tried on my daily use FF and the "Actual" is from Nightly.

**Comment by clouserw** [comment]
Oh, wow.  That's even stranger then.  Can you try going to accounts.firefox.com/clear in your Nightly browser and then try logging in again?  That will clear your browser side storage for FxA.

**Comment by siddhartharao17** [comment]
Sure, I did those steps and here are the screenshots. After clearing browser storage, clicked on the "Sign In" button for "Sync and save data" from the top right hamburger menu. Getting the "create account" screen instead of "login" screen. Seems like the email ID is not getting checked if it exists or not on the backend?

![Screen Shot 2022-09-14 at 10 19 17](https://user-images.githubusercontent.com/8169998/190180323-670ba5e8-8cf5-4d62-b1a2-7401cafe0c8b.png)
![Screen Shot 2022-09-14 at 10 19 43](https://user-images.githubusercontent.com/8169998/190180330-339456f9-648f-4080-a8c5-ff2cdee5a411.png)

**Comment by siddhartharao17** [comment]
I just tested what you suggested and I do see the email is being checked in the POST request. As I started to compile all this and email you, I realized there was a typo in my Nightly tests. I usually use my creds from my password manager but I probably typed my email in the other day and got it wrong, so that's my bad. 

There's no issue here anymore, I just tested with the rightly typed email in Nightly and it works perfectly! Sorry about all the trouble.

**Comment by clouserw** [comment]
whew 😅 I'm glad we got it figured out.  Thanks for the update!  Cheers
